### DISCUSSION TITLE
How can I enable --json parameter via webpack configuration file?

### DISCUSSION THREAD

**Discussion by stephane-klein** [webpack/webpack#16157] [Q&A]
Hi,

In a company project (I can't upgrade easily), I use `webpack@4.46.0` + `lerna@3.22.1` + `@craco/craco@6.4.3`.

I would like to set up [`bundle-stats`](https://github.com/relative-ci/bundle-stats) tooling in this environment.

The `bundle-stats` documentation, explain I must execute something like that:

```
npx webpack --mode production --json artifacts/webpack-stats.json
```

but [I don't know how to enable webpack `--json` option with `craco`](https://github.com/dilanx/craco/issues/391#issuecomment-1216751349).

**question:** How can I enable `--json` parameter via webpack [configuration file](https://v4.webpack.js.org/configuration/)?

Best regards,  
Stéphane

**Comment by stephane-klein** [comment]
I found these options:

- [`profile: true`](https://v4.webpack.js.org/configuration/other-options/#profile)
- [`recordsPath: path.join(__dirname, 'records.json')`](https://v4.webpack.js.org/configuration/other-options/#recordspath)

**Reply by stephane-klein** [reply]
> I found these options:
> 
>     * [`profile: true`](https://v4.webpack.js.org/configuration/other-options/#profile)
> 
>     * [`recordsPath: path.join(__dirname, 'records.json')`](https://v4.webpack.js.org/configuration/other-options/#recordspath)

Error, `records.json` isn't a stats file 😥

**Reply by alexander-akait** [reply]
You can't enable json using configuration file, I think it is a feature request to `craco`, also you can:
1. Create cusom runner for webpack (i.e. rerun webpack using spawn and pass request options)
2. We can accept in https://github.com/webpack/webpack-cli/ `WEBPACK_STATS_JSON=1` env variable

**Reply by stephane-klein** [reply]
Thanks for your response ♥

> We can accept in https://github.com/webpack/webpack-cli/ WEBPACK_STATS_JSON=1 env variable

@alexander-akait when you say « we can accept », you mean that it is a feature to develop? Not an existing option?

**Reply by alexander-akait** [reply]
@stephane-klein yes, it is a feature, feel free to send a PR, should not be hard (or open an issue)
